tests/framework/artifact/test_artifact.py

```diff
@@ -126,6 +127,27 @@ def test_artifact_load(hdf_mock, keys_mock):
         hdf_mock.load.reset_mock()
 
 
+def test_artifact_contains(hdf_mock, keys_mock):
+    path = Path('/place/with/artifact.hdf')
+    filter_terms = ['location == Global', 'draw == 10']
+    not_a_key = 'not.a.key'
+
+    art = Artifact(path, filter_terms)
+
+    for key in art.keys:
+        assert key in art
+        assert EntityKey(key) in art
```

Comment:
I don't think these EntityKey checks are necessary due to the fact that I'm explicitly testing how equality works for EntityKey in test_hdf.py, but I've included them here for now in case you want them. Without them, this test didn't fail before the EntityKey equality changes were made (which should be fine?)

Reply:
The failure cases were:

1. open artifact, write with entity key, load with raw string
2. open artifact, write with raw string, load with entity key
3. open artifact, write with raw string or entity key, close artifact, open artifact, load with entity key

Reply:
These are good regression tests, if nothing else, and they're cheap to run.